core/src/main/scala/org/apache/spark/internal/config/package.scala

```diff
@@ -809,6 +809,12 @@ package object config {
       .booleanConf
       .createWithDefault(false)
 
+  private[spark] val EXCLUDE_ON_FAILURE_DECOMMISSION_ENABLED =
+    ConfigBuilder("spark.excludeOnFailure.decommissionExcludedExecutors")
```

Comment:
Seems like this only takes effect when `spark.excludeOnFailure.killExcludedExecutors=true`. If so, I think it's more reasonable to have the same namespace for this conf, e.g., `spark.excludeOnFailure.killExcludedExecutors.decommissionEnabled` or something else.

And with the current implementation, the executor may not get killed if `EXECUTOR_DECOMMISSION_CLEANUP_INTERVAL` is not set, which then doesn't respect the `spark.excludeOnFailure.killExcludedExecutors=true` indeed. 

Actually, I'm thinking that it might be better to have these two confs at the same level?, e.g.,

```scala
if (EXCLUDE_ON_FAILURE_KILL_ENABLED) {
 // do kill
} else if (EXCLUDE_ON_FAILURE_DECOMMISSION_ENABLED) {
 // do decommission
}
// else do nothing
```

WDYT?

Reply:
I like the nested one because if they're both set to true in the second configuration priority is ambiguous.